pysatNASA/instruments/timed_guvi.py

```diff
@@ -0,0 +1,233 @@
+                'min_val': ('value_min', float),
+                'max_val': ('value_max', float),
+                'fill_val': ('fill', float)}
+    meta = pysat.Meta(labels=labels)
+
+    daysubset,nightsubset,auroralsubset=None,None,None
+    for path in fnames:
+        dataset=xr.load_dataset(path)
+        
+        #separate into auroral, night and day datasets
+        auroralkeys=list(filter(lambda k:"_AURORAL" in k[-8:],dataset.keys()))
+        nightkeys=list(filter(lambda k:"_NIGHT" in k,dataset.keys()))
+        daykeys=list(filter(lambda k:"_DAY" in k and "_AURORAL" not in k,dataset.keys()))
+        otherkeys=list( filter( lambda k:k not in (daykeys+nightkeys+auroralkeys),dataset.keys() ) )
+        
+        ith_daysubset=dataset.drop_vars(auroralkeys+nightkeys)
+        ith_nightsubset=dataset.drop_vars(auroralkeys+daykeys+otherkeys)
+        ith_auroralsubset=dataset.drop_vars(daykeys+nightkeys+otherkeys)
+        
+        #Updating epoch time variables, and dropping redundant time variables year, doy, day
+        dts=[datetime(year,1,1)+timedelta(days=float(ith_daysubset.DOY_DAY.data[i]-1),seconds=ith_daysubset.TIME_DAY.data[i]) for i,year in enumerate(ith_daysubset.YEAR_DAY.data)]
+        ith_daysubset=ith_daysubset.drop_vars(["YEAR_DAY","DOY_DAY","TIME_DAY","TIME_EPOCH_DAY"])
+        ith_daysubset=ith_daysubset.assign( { 'TIME_EPOCH_DAY': xr.DataArray(data=dts,dims=('nAlongDay')) } )
+        
+        dts=[datetime(year,1,1)+timedelta(days=float(ith_nightsubset.DOY_NIGHT.data[i]-1),seconds=ith_nightsubset.TIME_NIGHT.data[i]) for i,year in enumerate(ith_nightsubset.YEAR_NIGHT.data)]
+        ith_nightsubset=ith_nightsubset.drop_vars(["YEAR_NIGHT","DOY_NIGHT","TIME_NIGHT",'TIME_EPOCH_NIGHT'])
+        ith_nightsubset=ith_nightsubset.assign( { 'TIME_EPOCH_NIGHT': xr.DataArray(data=dts,dims=('nAlongNight')) } )
+        
+        dts=[datetime(year,1,1)+timedelta(days=float(ith_auroralsubset.DOY_DAY_AURORAL.data[i]-1),seconds=ith_auroralsubset.TIME_DAY_AURORAL.data[i]) for i,year in enumerate(ith_auroralsubset.YEAR_DAY_AURORAL.data)]
+        ith_auroralsubset=ith_auroralsubset.drop_vars(["YEAR_DAY_AURORAL","DOY_DAY_AURORAL","TIME_DAY_AURORAL","TIME_EPOCH_DAY_AURORAL"])
+        ith_auroralsubset=ith_auroralsubset.assign( { 'TIME_EPOCH_DAY_AURORAL': xr.DataArray(data=dts,dims=('nAlongDayAur')) } )
+        
+        #concatenate along corresponding dimension
+        if daysubset is None:
+            daysubset=ith_daysubset
+            nightsubset=ith_nightsubset
+            auroralsubset=ith_auroralsubset
+        else:
+            daysubset=xr.concat([daysubset,ith_daysubset],dim='nAlongDay')
+            nightsubset=xr.concat([nightsubset,ith_nightsubset],dim='nAlongNight')
+            auroralsubset=xr.concat([auroralsubset,ith_auroralsubset],dim='nAlongDayAur')
+    
+    #merge all datasets
+    data=daysubset
+    data=data.merge(nightsubset)
+    data=data.merge(auroralsubset)
+
```

Comment:
I can confirm the data loads. guvi reports as `empty` since pysat checks against indexes with the xarray. In this case it doesn't find any. This is the source of the `clean` test failure. pyast also requires that the main time dimension be named 'time'.

In our initial conversations I was expecting the GUVI file to be a CDF. I see now that it is netCDF. The load methods employed for ICON could be useful here. Note there are some improvements coming in the pipe, https://github.com/pysat/pysat/pull/983 and some of them have been used on ICON, https://github.com/pysat/pysatNASA/pull/100 

Related to this, there is no information in the guvi metadata. The load that ICON employs should also handle that part for you. 

COSMIC GPS over in pysatCDAAC `develop` loads profile data into xarray. It can serve as another example.

```
In [18]: gps = pysat.Instrument("cosmic", "gps", "ionprf")

In [19]: gps.load(2008, 1)
<ipython-input-19-bb6939bfe4e3>:1: DeprecationWarning: Meta now contains a class for global metadata (MetaHeader). Default attachment of global attributes to Instrument will be Deprecated in pysat 3.2.0+. Set `use_header=True` to remove this warning.
  gps.load(2008, 1)

In [20]: gps.empty
Out[20]: False

In [21]: gps.data
Out[21]: 
<xarray.Dataset>
Dimensions:           (time: 1199, RO: 669)
Coordinates:
  * time              (time) datetime64[ns] 2008-01-01T00:08:16.000084 ... 20...
    MSL_alt           (time, RO) float64 51.46 54.22 56.97 59.72 ... nan nan nan
    GEO_lat           (time, RO) float64 34.3 34.32 34.33 34.35 ... nan nan nan
    GEO_lon           (time, RO) float64 -85.61 -85.62 -85.63 ... nan nan nan
    OCC_azi           (time, RO) float64 131.1 131.0 130.9 130.8 ... nan nan nan
Dimensions without coordinates: RO
Data variables: (12/53)
    occ_id            (time) float64 1.0 nan 1.0 nan nan ... nan 1.0 1.0 nan 1.0
    fiducial_id       (time) object '' nan '' nan nan '' ... '' nan '' '' nan ''
    reference_sat_id  (time) float64 0.0 nan 0.0 nan nan ... nan 0.0 0.0 nan 0.0
    occulting_sat_id  (time) float64 8.0 nan 31.0 nan nan ... 4.0 21.0 nan 8.0
    year              (time) float64 2.008e+03 nan 2.008e+03 ... nan 2.008e+03
    month             (time) float64 1.0 nan 1.0 nan nan ... nan 1.0 1.0 nan 1.0
    ...                ...
    parmsfile         (time) object 'parms8' nan 'parms8' ... nan 'parms8'
    center            (time) object 'UCAR/CDAAC' nan ... nan 'UCAR/CDAAC'
    mission           (time) object 'COSMIC' nan 'COSMIC' ... nan 'COSMIC'
    creation_time     (time) object '10-JUL-14 18:56' nan ... '10-JUL-14 18:56'
    TEC_cal           (time, RO) float64 19.15 19.44 17.18 17.38 ... nan nan nan
    ELEC_dens         (time, RO) float64 nan nan nan nan nan ... nan nan nan nan

In [22]: gps.data.indexes
Out[22]: 
time: DatetimeIndex(['2008-01-01 00:08:16.000084', '2008-01-01 00:11:04.000202',
                     '2008-01-01 00:11:29.000312', '2008-01-01 00:12:20.000274',
                     '2008-01-01 00:15:45.000254', '2008-01-01 00:23:33.000041',
                     '2008-01-01 00:26:48.000302', '2008-01-01 00:27:07.000082',
                     '2008-01-01 00:29:51.000052', '2008-01-01 00:31:28.000121',
                     ...
                     '2008-01-01 23:49:01.000283', '2008-01-01 23:50:30.000124',
                     '2008-01-01 23:50:40.000144', '2008-01-01 23:51:28.000054',
                     '2008-01-01 23:52:26.000304', '2008-01-01 23:54:01.000133',
                     '2008-01-01 23:54:39.000041', '2008-01-01 23:55:55.000212',
                     '2008-01-01 23:56:14.000171', '2008-01-01 23:57:07.000082'],
                    dtype='datetime64[ns]', name='time', length=1199, freq=None)
```